### DISCUSSION TITLE
Docker install in homelab (behind CGNAT) Cloudflare Tunnel?

### DISCUSSION THREAD

**Discussion by onehumankindness** [Peppermint-Lab/peppermint#227] [Q&A] [answered]
Has anyone used Cloudflare zero trust tunnel to make a homelab install of Peppermint accessible online? If so, what's the trick I tried setting it up with and without /auth/login path and no workie :(

**Comment by thackmaster** [comment, marked as answer]
I haven't seen issues here. I just tested on a fresh install, and I have not issues. Unless you're using a reverse proxy in the middle somewhere, this should work out for you.
<img width="1050" alt="Screenshot 2024-03-29 at 16 05 56" src="https://github.com/Peppermint-Lab/peppermint/assets/73045785/2e1cffe7-d79e-4631-a612-32b05fca16cb">

**Reply by jwaresolutions** [reply]
I am having the same issue. fresh install, no changes made. I can get to the server by IP, just not through the Cloudflare tunnel setup exactly like the above screenshot.

**Reply by thackmaster** [reply]
There are a few things that come to mind that I'm going to assume you've checked but just for sanity sake:

1. Make sure the port listed for the tunnel is the same port you exposed
2. Make sure cloudflared is running on the host
3. Check the logs of cloudflared and make sure it's receiving authorization to pass traffic through
4. Are there other tunnels to the same host you have exposed? If yes, do they work with the tunnel?

**Reply by jwaresolutions** [reply]
I got it working. I had originally used the marketplace version in linode, but that was attempting to create more than i needed. So i just deployed from a docker-compose file and everything worked just fine.

**Reply by jwaresolutions** [reply]
So in my case the problem was #4
